hw/app.rb

```diff
@@ -0,0 +1,55 @@
+require 'sinatra'
+require 'sinatra/reloader'
+
+
+get '/' do
+  erb :index
+end
+
+# - The homepage should list several menu options:
+#   * List all apartments (a link to `GET /apartments`)
+
+
+get '/apartments' do
+  erb :apartment_index
+end
+
+#   * View an apartment's details(a link to `GET /apartments/1`)
+get 'apartments/:id' do
```

Comment:
You'll want to move your `apartments/:id` route below `apartments/new`. Otherwise, your controller is always going to interpret "new" as an apartment id, which will always return an error because "new" is a string.